cassandra/cqlengine/query.py

```diff
@@ -819,12 +819,19 @@ class Comment(Model):
             clone._order = []
             return clone
 
-        conditions = []
-        for colname in colnames:
-            conditions.append('"{0}" {1}'.format(*self._get_ordering_condition(colname)))
-
         clone = copy.deepcopy(self)
-        clone._order.extend(conditions)
+
+        for col_name in colnames:
+            col_name, order = self._get_ordering_condition(col_name)
+            col_name = '"{0}"'.format(col_name)
+            # Check if there is existing condition that targets same column and replace it if it's order is different
+            for cond_id, existing_condition in enumerate(clone._order):
+                if col_name in existing_condition:
+                    if not existing_condition.endswith(order):
+                        clone._order[cond_id] = '{0} {1}'.format(col_name, order)
+                    break
+            else:
+                clone._order.append('{0} {1}'.format(col_name, order))
```

Comment:
Question regarding commit "Fix AbstractQuerySet.order_by to consider existing conditions ":
The root cause of the problem is that Scylla handles such weird queries differently than Cassandra, right? If so, could you add a comment on this code that explains it?

Note: I know nothing about cqlengine code so I don't feel competent to review it and I don't feel like I understand the change. @fruch can you take a look?

Reply:
this fix not related to cqlengine per se, 
seems it cover from some mismatch between scylla and cassandra 

it's weird a bit, cassandra accepts this ? what the CQL spec say about it ? 

seems not much:
https://cassandra.apache.org/doc/4.0/cassandra/cql/dml.html#ordering-clause

I think we should raise it as issue in scylla, and mark it as compatibility issue with cassandra

Reply:
That's something the server should be doing/guarding, but for the meanwhile, it's seems o.k. to me that that cqlengine would prevent it.

Reply:
Problem is that scylla does that and fails the query with multiple records of the same column in `order by` while cassandra ignores it and goes with order mentioned last.

Reply:
> Problem is that scylla does that and fails the query with multiple records of the same column in `order by` while cassandra ignores it and goes with order mentioned last.

And that fix makes cqlengine avoid creating such queries, so they are basically ignored like in cassandra, even when executing on Scylla?
Is it the right thing to do here? It fixes the test, but isn't the Scylla behavior better for the user (failing early rather than somehow executing it)?

What do you think about changing / xfailing the test instead?

Reply:
> And that fix makes cqlengine avoid creating such queries, so they are basically ignored like in cassandra, even when executing on Scylla? Is it the right thing to do here? 
Yes.

It fixes the test, but isn't the Scylla behavior better for the user (failing early rather than somehow executing it)?
Yes, I think scylla behavior is more correct, it does not make sense to have opposite ordering direction for the same column.

> What do you think about changing / xfailing the test instead?
As far as understand this test, this is `cqlengine` test for `order_by` functionality that generates one query executes it and then reverse order direction and executes it again, so it should work just fine.

Reply:
Ok, after thinking about it just a bit, I decided to fix the test instead, commit is dropped.